### PR TITLE
fix(core): deduplicate union type members in generated code

### LINKED ISSUE #2852: Generated types contain duplicate union members (e.g., `null | null`, `Type | Type`)
### Description

Generated types can contain duplicate union members. While these duplicates don't affect runtime behavior, they create noise in generated code. I have a working fix ready to submit as a PR.

### Configuration (orval.config)

[`samples/react-query/basic/orval.config.ts`](https://github.com/orval-labs/orval/blob/master/samples/react-query/basic/orval.config.ts)

### Environment

```
System:
  OS: macOS 15.3.1
  CPU: (12) arm64 Apple M4 Pro
  Memory: 63.89 MB / 24.00 GB
  Shell: 5.9 - /bin/zsh
npmPackages:
  @tanstack/react-query: ^5.74.3 => 5.74.7 
  axios: ^1.8.4 => 1.12.2 
  orval: ^8.0.3 => 8.0.3 
  react: 18.3.1 => 18.3.1 
  zod: ^3.24.1 => 3.25.76 
```

### Expected behavior

```typescript
// Nullable object
export type NullableObject = { [key: string]: unknown } | null;

// Multiple success responses referencing same schema
result.definition.success === 'Pet'

// Multiple error responses referencing same schema
result.definition.errors === 'Error'

// oneOf with duplicate types
export type DuplicateUnion = string | number;
```

### Actual behavior

```typescript
// Nullable object - duplicate null
export type NullableObject = { [key: string]: unknown } | null | null;

// Multiple success responses - duplicate type
result.definition.success === 'Pet | Pet'

// Multiple error responses - duplicate type
result.definition.errors === 'Error | Error | Error'

// oneOf with duplicate types - duplicate string
export type DuplicateUnion = string | string | number;
```

### OpenAPI document (minimal, if applicable)

```json
{
  "components": {
    "schemas": {
      "NullableObject": {
        "type": ["object", "null"]
      },
      "DuplicateUnion": {
        "oneOf": [
          { "type": "string" },
          { "type": "string" },
          { "type": "number" }
        ]
      }
    }
  },
  "paths": {
    "/pets": {
      "post": {
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": { "$ref": "#/components/schemas/Pet" }
              }
            }
          },
          "201": {
            "content": {
              "application/json": {
                "schema": { "$ref": "#/components/schemas/Pet" }
              }
            }
          },
          "400": {
            "content": {
              "application/json": {
                "schema": { "$ref": "#/components/schemas/Error" }
              }
            }
          },
          "401": {
            "content": {
              "application/json": {
                "schema": { "$ref": "#/components/schemas/Error" }
              }
            }
          },
          "404": {
            "content": {
              "application/json": {
                "schema": { "$ref": "#/components/schemas/Error" }
              }
            }
          }
        }
      }
    }
  }
}
```

### Additional context

**Related:**
- #2289
- #2177

### PR DESCRIPTION
**Pull request by ophirbucai**
## Summary

Fixes #2852 - Generated types no longer contain duplicate union members.

**Before:** `{ [key: string]: unknown } | null | null`, `Pet | Pet`
**After:** `{ [key: string]: unknown } | null`, `Pet`

## Changes

Deduplicate at the output level by splitting union strings, removing duplicates with Set, and rejoining.

- Add `dedupeUnionType` utility in `string.ts`
- Apply in `response.ts`: dedupe success/error type definitions
- Apply in `combine.ts`: dedupe combined schema values with nullable

## Related

- #2289
- #2177

## Tested

- `yarn build` ✓
- `yarn test` ✓
- `yarn update-samples` ✓
- `yarn test:samples` ✓
- `yarn test:cli` ✓
- Added tests for nullable objects, oneOf duplicates, and duplicate response types
- Added unit tests for dedupeUnionType edge cases (empty string, whitespace, order preservation)

### CODE REVIEW

File: packages/core/src/utils/string.ts [hunk 1]
```diff
@@ -251,3 +251,15 @@ export function jsStringEscape(input: string) {
...
 }
+
+/**
+ * Deduplicates a TypeScript union type string.
+ * Handles types like "A | B | B" → "A | B" and "null | null" → "null".
+ */
+export function dedupeUnionType(unionType: string): string {
+  if (!unionType) return unionType;
+
+  const parts = unionType.split('|').map((part) => part.trim());
```
[the remark this answers is not in the document]
Comment (ophirbucai) [reply packages/core/src/utils/string.ts:260]: I think this is overkill for the library's usage - which effectively handles simple unions. JSON outputs do not have `|` in them, as they are represented in OpenAPI as `oneOf`.
